SecuritySystemUWP/SecuritySystemUWP/Storage/OneDrive.cs

```diff
@@ -158,5 +154,98 @@ private void SaveTokens(object sender, string arg)
             App.Controller.XmlSettings.OneDriveAccessToken = oneDriveConnector.accessToken;
             App.Controller.XmlSettings.OneDriveRefreshToken = oneDriveConnector.refreshToken;
         }
+
+        private async Task uploadWithRetry(StorageFile file, int tryNumber)
+        {
+            HttpResponseMessage response = await oneDriveConnector.UploadFileAsync(file, String.Format("{0}/{1}", App.Controller.XmlSettings.OneDriveFolderPath, DateTime.Now.ToString("yyyy_MM_dd")));
+            bool success = await parseResponse(response, tryNumber);
+            var events = new Dictionary<string, string>();
+
+            if (success)
+            {
+                numberUploaded++;
+                await file.DeleteAsync();
+                this.lastUploadTime = DateTime.Now;
+            }
+            else if (tryNumber <= MaxTries)
+            {
+                events.Add("Retrying upload", tryNumber.ToString());
+                TelemetryHelper.TrackEvent("FailedToUploadPicture", events);

```

Comment:
Can we rename this event to "FailedToUploadPicture Recovery Attempt" so we can distinguish it from the failure event?